ocaml/xapi/xapi_sdn_controller.ml

```diff
@@ -0,0 +1,54 @@
+(*
+ * Copyright (C) 2006-2016 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Stdext
+open Pervasiveext
+open Xstringext
+open Helpers
+open Listext
+open Client
+open Stdext.Threadext
+open Unixext
+open Network
+
+module D = Debug.Make(struct let name="xapi" end)
+open D
+
+let protocol_to_string = function
+  | `ssl -> "ssl"
+  | `pssl -> "pssl"
+  | _ -> "ssl"
+
+let introduce ~__context ~protocol ~address ~port =
+  let dbg = Context.string_of_task __context in
+  match Net.Bridge.get_kind dbg () with
+  | Network_interface.Openvswitch ->
+    if List.length (Db.SDN_controller.get_all ~__context) > 0 then
+      raise (Api_errors.Server_error(Api_errors.operation_not_allowed, ["SDN controller has been configured. Please forget it first."]));
+      if address <> "" then
+        Helpers.assert_is_valid_ip `ipv4 "address" address;
+      if port <> 0L then
```

Comment:
As `0` is the port default if client doesn't set any value, is it reasonable we set it to `6632` if protocol is `ssl`?